mdbook-mermaid needs updating

`mdbook-mermaid` is locked to 0.6.1 in the CI build, which is failing on nightly because of yanked packages:

[Failed build](https://github.com/rust-lang/chalk/runs/7975830842?check_suite_focus=true#step:8:16)
```
cargo install mdbook-mermaid --locked --version 0.6.1
 Downloading crates ...
  Downloaded mdbook-mermaid v0.6.1
    Updating crates.io index
  Installing mdbook-mermaid v0.6.1
warning: package `ammonia v3.1.0` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `futures v0.3.5` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `futures-util v0.3.5` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `miow v0.2.1` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `net2 v0.2.33` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `pin-project v0.4.24` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `pin-project-lite v0.1.8` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `socket2 v0.3.15` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
...
   Compiling socket2 v0.3.15
error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/socket2-0.3.15/src/sockaddr.rs:176:9
    |
176 |         mem::transmute::<SocketAddrV4, sockaddr_in>(v4);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: source type: `SocketAddrV4` (48 bits)
    = note: target type: `sockaddr_in` (128 bits)
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -47,7 +47,7 @@ jobs:
           echo "$GITHUB_WORKSPACE/book" >> $GITHUB_PATH
 
       - name: Install mdbook-mermaid
-        run: cargo install mdbook-mermaid --locked --version 0.6.1
+        run: cargo install mdbook-mermaid
 
       - name: Execute tests for Chalk book
         run: cd book && ./mdbook test

